File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        
        unordered_map<char,int>gzip,order;
        
        gzip['I']=1;
        gzip['V']=5;
        gzip['X']=10;
        gzip['L']=50;
        gzip['C']=100;
        gzip['D']=500;
        gzip['M']=1000;
        
        order['I']=1;
        order['V']=2;
        order['X']=3;
        order['L']=4;
        order['C']=5;
        order['D']=6;
        order['M']=7;
        
        int i=0;
        int ans=0;
        while(i<s.length()){
                
//                 if(i+1>=s.length()){
                
//                         ans+=gzip[s[i]];
//                         i+=1;
                    
//                 }    
            
                if(order[s[i]]>=order[s[i+1]]){
                        
                        ans+=gzip[s[i]];
                        i+=1;  
                }
                else{
                        ans+=(gzip[s[i+1]]-gzip[s[i]]);
                        i+=2;
                    
                }
            
              
            
        }
        
        return ans;
        
    }
};

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        
            d={}
            d['I']=1
            d['V']=5
            d['X']=10
            d['L']=50
            d['C']=100
            d['D']=500
            d['M']=1000
        
            sum=0
            for i in reversed(range(len(s))):
                
                
                if i<len(s)-1 and d[s[i]]<d[s[i+1]]:
                        sum-=d[s[i]]
                        
                else:
                        sum+=d[s[i]]
                 
               
             
            return sum
                
                